internal/handlers/common/update.go

```diff
@@ -899,10 +899,69 @@ func newUpdateError(code commonerrors.ErrorCode, msg, command string) error {
 	return commonerrors.NewWriteErrorMsg(code, msg)
 }
 
+// newPathValidator creates a root of pathValidator.
+// It shouldn't be used as a node of other pathValidator.
+func newPathValidator() *pathValidator {
+	return &pathValidator{
+		paths: map[string]*pathValidator{},
+		root:  true,
+	}
+}
+
+// pathValidator stores validated paths and checks for any conflicts
+// between them. The conflict is when one path points to the part of
+// another path.
+//
+// Examples of conflicts:
+// `foo.bar, foo.bar`
+// `foo.bar, foo.bar.x`
+//
+// This is not a conflict:
+// `foo.bar.x, foo.bar.y`.
+type pathValidator struct {
+	paths map[string]*pathValidator
+	last  bool
+	root  bool
+}
+
+// validate takes a slice of path keys and checks if the path
+// conflicts with previously validated path. If it's not, then
+// the path is saved in pathValidator.
+func (tree *pathValidator) validate(keys []string) bool {
```

Comment:
Does this function work if we have the same key on the path? `{foo: {foo: {foo:}}}` and trying to insert `foo.foo.bar` for example?

Reply:
`foo.foo.foo` does not conflict with `foo.foo.bar`. Both of them conflicts with `foo.foo` though. That's how validator should work here.

Reply:
I initially didn't understand that the `pathValidator` was created for each node, so this one is clear now!